src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -3033,6 +3037,17 @@ class WithNoSchemaBindingClauseSegment(BaseSegment):
     )
 
 
+@ansi_dialect.segment()
+class WithDataClauseSegment(BaseSegment):
+    """WITH [NO] DATA clause for Postgres' MATERIALIZED VIEWS.
+
+    https://www.postgresql.org/docs/9.3/sql-creatematerializedview.html
+    """
+
+    type = "with_no_schema_binding_clause"
```

Comment:
```suggestion
    type = "with_data_clause"
```
We can fix that type I missed here